graph-gateway/src/config.rs

```diff
@@ -65,10 +65,13 @@ pub struct Config {
     /// Minimum indexer-service version that will receive queries
     #[serde_as(as = "DisplayFromStr")]
     pub min_indexer_version: Version,
-    /// Network subgraph query path
-    #[debug(with = Display::fmt)]
-    #[serde_as(as = "DisplayFromStr")]
-    pub network_subgraph: Url,
+    /// Network subgraph discovery service configuration
+    pub network: NetworkSubgraphServiceConfig,
+    /// Network subgraph url
+    #[deprecated(note = "Use `network` instead")]
+    #[debug(with = fmt_debug_optional_url)]
+    #[serde_as(as = "Option<DisplayFromStr>")]
```

Comment:
I kept this for backward configuration compatibility. But, shall we remove this configuration directly?

Reply:
Yes. This will be part of a major release.